src/join.rs

```diff
@@ -155,6 +155,61 @@ pub trait Join {
         JoinIter::new(self)
     }
 
+    /// Optionally gets a `Join` if it exists.
```

Comment:
This sentence sounds confusing to me. Suggesting: `Returns a \`Join\`able` that yields for all indices, returning `None` for all missing elements.`